broadlink/switch.py

```diff
@@ -221,35 +221,35 @@ def get_energy(self) -> int:
 class sp4(device):
     """Controls a Broadlink SP4."""
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, **kwargs) -> None:
         """Initialize the controller."""
         device.__init__(self, *args, **kwargs)
         self.type = "SP4"
 
-    def set_power(self, state):
+    def set_power(self, state: bool) -> bool:
```

Comment:
These functions should return None. If something goes wrong the device will return an error in the response and we raise [here](https://github.com/mjg59/python-broadlink/blob/d8b853d96c9bc64a253e41657789359f6bb165bd/broadlink/switch.py#L305). So there is no reason to return a state we already know. Follow the SP2 template.